shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -894,6 +890,18 @@ protected FlutterEngine getFlutterEngine() {
     return delegate.getFlutterEngine();
   }
 
+  /** Retrieves the meta data specified in the AndroidManifest.xml. */
+  @Nullable
+  protected Bundle getMetaData() throws RuntimeException {
```

Comment:
Ah sorry my bad. I hadn't realized it wasn't public and just noticed now that you changed places where it's reading this and they're already catching PackageManager.NameNotFoundException. In that case, I think we can leave it as it was. With this method throwing PackageManager.NameNotFoundException(and in the fragment activity too) and letting places that call it catch PackageManager.NameNotFoundException and do a reasonable fallback. Sorry I misread.